### PR TITLE
feat(cli): support walk vue and astro

### PR DESCRIPTION
**Pull request by mysteryven**

### CODE REVIEW

**Comment by mysteryven** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1745** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1745?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1745?utm_source=stack-comment).

File: crates/oxc_linter/src/lib.rs [hunk 1]
```diff
@@ -44,6 +44,19 @@ impl Default for LintSettings {
...
 
+/// Extract the JS part from the source code.
+#[derive(Clone)]
+pub enum LintPartialLoader {
+    Vue,
+    Astro,
+}
+
+impl LintPartialLoader {
+    pub fn is_support(ext: &str) -> bool {
```
Comment (Boshen) [inline crates/oxc_linter/src/lib.rs:55]: `filter_extension`?

File: crates/oxc_linter/src/lib.rs [hunk 2]
```diff
@@ -44,6 +44,19 @@ impl Default for LintSettings {
     }
 }
 
+/// Extract the JS part from the source code.
+#[derive(Clone)]
+pub enum LintPartialLoader {
```
Comment (Boshen) [inline crates/oxc_linter/src/lib.rs:49]: Can we create a separate file for this?

File: crates/oxc_cli/src/walk.rs [hunk 3]
```diff
@@ -108,6 +109,8 @@ impl Walk {
             return false;
         }
         let Some(extension) = dir_entry.path().extension() else { return false };
-        VALID_EXTENSIONS.contains(&extension.to_string_lossy().as_ref())
+        let extension = extension.to_string_lossy();
+        VALID_EXTENSIONS.contains(&extension.as_ref())
```
[the remark this answers is not in the document]
Comment (mysteryven) [reply crates/oxc_cli/src/walk.rs:113]: I wrote like this at first time, is these loaders are common for our linter and Formater?

File: crates/oxc_cli/src/walk.rs [hunk 4]
```diff
@@ -8,37 +8,55 @@ use oxc_span::VALID_EXTENSIONS;
 
 use crate::IgnoreOptions;
 
-pub struct Walk {
+#[derive(Clone, Copy)]
+pub struct Extensions<'a>(pub &'a [&'a str]);
```
Comment (Boshen) [inline crates/oxc_cli/src/walk.rs:12]: `Vec<'static str>`

**Review by Dunqing — APPROVED** [review]

**Comment by Boshen** [comment]
### Merge activity

* **Dec 31, 9:14 AM**: @Boshen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1745).

**Pull request merged**